### PR TITLE
chore: add dev docs for swift sdk

### PR DESCRIPTION
**Pull request by anitarua**
Swift dev docs to address https://github.com/momentohq/client-sdk-swift/issues/56

### CODE REVIEW

**Comment by anitarua** [comment]
@poppoerika I'm not sure how to get the Translate CI job to pass. I copied the new files to the Japanese translation section as well, but I can remove them if that's not necessary

File: i18n/ja/docusaurus-plugin-content-docs/current/topics/develop/sdks/swift/cheat-sheet.mdx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+You'll need a Momento API key. You can get one from the [Momento Web Console](https://console.gomomento.com/).
+Once you have your token, store it in an environment variable so that the Momento client can consume it:
+
+```
+export MOMENTO_API_KEY=<your api key here>
+```
+
+**Note**: it is best practice to put the API key into something like AWS Secret Manager or GCP Secret Manager instead of an environment variable for enhanced security, but we are using one here for demo purposes.
+
+## Set up a CacheClient
```
Comment (pgautier404) [inline i18n/ja/docusaurus-plugin-content-docs/current/topics/develop/sdks/swift/cheat-sheet.mdx]:
```suggestion
## Set up a TopicClient
```

File: i18n/ja/docusaurus-plugin-content-docs/current/topics/develop/sdks/swift/cheat-sheet.mdx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+This code creates the TopicClient that you will use to call the other methods.
+
+<SdkExampleCodeBlock language={'swift'} snippetId={'API_InstantiateTopicClient'} />
+
+## Publish a message to a topic
+A simple example of doing a set operation. In the CacheClient.set() call, the TTL is optional. If you did pass it in, this would override the default TTL value set with the client connection object.
```
Comment (pgautier404) [inline i18n/ja/docusaurus-plugin-content-docs/current/topics/develop/sdks/swift/cheat-sheet.mdx]: Looks like some escaped copypasta :-)

File: i18n/ja/docusaurus-plugin-content-docs/current/topics/develop/sdks/swift/cheat-sheet.mdx [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+<SdkExampleCodeBlock language={'swift'} snippetId={'API_Publish'} />
+
+## Subscribe to a topic
+This is an example of a simple read operation to get an item from a cache.
```
Comment (pgautier404) [inline i18n/ja/docusaurus-plugin-content-docs/current/topics/develop/sdks/swift/cheat-sheet.mdx]: Same here.

**Review by pgautier404 — COMMENTED** [review]
We're not getting any syntax highlighting for the Swift code. @cprice404 is there something we need to do to enable that?

It also looks like the snippets in the API reference for `publish` and `subscribe` aren't showing their tabs. The topic client snippet is showing up fine, though it is also missing syntax highlighting.

The last thing I noticed was there are no snippets for the list operations, but since there are currently only Javascript snippets for those APIs anyhow, that doesn't really seem like a blocker.

File: docs/topics/develop/sdks/swift/cheat-sheet.mdx [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+You'll need a Momento API key. You can get one from the [Momento Web Console](https://console.gomomento.com/).
+Once you have your token, store it in an environment variable so that the Momento client can consume it:
+
+```
+export MOMENTO_API_KEY=<your api key here>
+```
+
+**Note**: it is best practice to put the API key into something like AWS Secret Manager or GCP Secret Manager instead of an environment variable for enhanced security, but we are using one here for demo purposes.
+
+## Set up a CacheClient
```
Comment (pgautier404) [inline docs/topics/develop/sdks/swift/cheat-sheet.mdx]:
```suggestion
## Set up a TopicClient
```

File: docs/topics/develop/sdks/swift/cheat-sheet.mdx [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+<SdkExampleCodeBlock language={'swift'} snippetId={'API_InstantiateTopicClient'} />
+
+## Publish a message to a topic
+A simple example of doing a set operation. In the CacheClient.set() call, the TTL is optional. If you did pass it in, this would override the default TTL value set with the client connection object.
+
+<SdkExampleCodeBlock language={'swift'} snippetId={'API_Publish'} />
+
+## Subscribe to a topic
+This is an example of a simple read operation to get an item from a cache.
```
Comment (pgautier404) [inline docs/topics/develop/sdks/swift/cheat-sheet.mdx]: These descriptions are for cache client :-)

**Comment by cprice404** [comment]
> We're not getting any syntax highlighting for the Swift code. @cprice404 is there something we need to do to enable that?
> 
> It also looks like the snippets in the API reference for `publish` and `subscribe` aren't showing their tabs. The topic client snippet is showing up fine, though it is also missing syntax highlighting.
> 
> The last thing I noticed was there are no snippets for the list operations, but since there are currently only Javascript snippets for those APIs anyhow, that doesn't really seem like a blocker.

you will want to check this out:

https://github.com/momentohq/public-dev-docs/blob/8dd9c03c6fb9784fd8d33b60de7730a60c55b99c/docusaurus.config.js#L371

You may have to look in the docusaurus (or related plugin) docs to see if swift is supported, but if it is, it will almost certainly need to be added there.

I haven't reviewed this PR, but if there are any other issues with the snippets showing up then there may be a few things we need to add/fix on the react components and associated plugin .ts files to make sure we've added the new language.

I agree that none of these necessarily need to block this PR, fine to make incremental progress.

**Comment by anitarua** [comment]
Added swift to the docusaurus config list and syntax highlighting works now.

The snippet issues should be fixed once this goes in (it was due to mismatch in snippet IDs): https://github.com/momentohq/client-sdk-swift/pull/113

Made a ticket to add list collection type snippets in future: https://github.com/momentohq/client-sdk-swift/issues/114

File: i18n/ja/docusaurus-plugin-content-docs/current/topics/develop/sdks/swift/cheat-sheet.mdx [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+## Publish a message to a topic
+This is an example of publishing a message to a topic called "topic", then catching the return value to check if the publish was successful.
+
+<SdkExampleCodeBlock language={'swift'} snippetId={'API_TopicPublish'} />
+
+## Subscribe to a topic
+This is an example of subscribing to a topic called "topic", then catching the return value to check if it was successful. If a subscription was received, a `for try await` loop is used to asynchronously receive and print the messages that are published to this topic.
+
+<SdkExampleCodeBlock language={'swift'} snippetId={'API_TopicSubscribe'} />
```
Comment (pgautier404) [inline i18n/ja/docusaurus-plugin-content-docs/current/topics/develop/sdks/swift/cheat-sheet.mdx:46]: These are showing up blank for me. Looks like they need to be `API_Publish` and `API_Subscribe` instead,

Comment (pgautier404) [reply i18n/ja/docusaurus-plugin-content-docs/current/topics/develop/sdks/swift/cheat-sheet.mdx:46]: Ah, nevermind, I see the fix for this in the other PR.

**Comment by anitarua** [comment]
@cprice404 do you know how I might be able to get the Japanese translation task to pass? or possibly disable it for this PR or something?

**Review by pgautier404 — APPROVED** [review]

**Pull request merged**